lib/web_ui/lib/src/engine/canvaskit/layer.dart

```diff
@@ -518,10 +533,15 @@ class ColorFilterEngineLayer extends ContainerLayer
     final CkPaint paint = CkPaint();
     paint.colorFilter = filter;
 
+    paintContext.internalNodesCanvas.save();
```

Comment:
Maybe leave a comment explaining why clipping is needed (and also leave a link to the issue that this fixed).